src/upgrades.ts

```diff
@@ -72,33 +76,40 @@ export async function deployProxy<T extends Contract>(
     opts?.proxyOpts
   )) as T
 
+  const deploymentTransaction = contractInstance.deploymentTransaction()
+
   // Let the transaction propagate across the ethereum nodes. This is mostly to
   // wait for all Alchemy nodes to catch up their state.
-  await contractInstance.deployTransaction.wait(1)
+  const transactionReceipt = await deploymentTransaction?.wait(1)
+
+  const contractAddress = await contractInstance.getAddress()
+  const transactionHash = deploymentTransaction?.hash
 
   log(
-    `Deployed ${name} as ${opts?.proxyOpts?.kind || "transparent"} proxy at ${
-      contractInstance.address
-    } (tx: ${contractInstance.deployTransaction.hash})`
+    `Deployed ${name} as ${
+      opts?.proxyOpts?.kind || "transparent"
+    } proxy at ${contractAddress} (tx: ${transactionHash})`
   )
 
   const artifact = artifacts.readArtifactSync(opts?.contractName || name)
 
-  const adminInstance = await upgrades.admin.getInstance()
-  const implementation = await adminInstance.getProxyImplementation(
-    contractInstance.address
+  const implementation = await upgrades.erc1967.getImplementationAddress(
+    contractAddress
   )
 
-  const transactionReceipt = await ethers.provider.getTransactionReceipt(
-    contractInstance.deployTransaction.hash
-  )
+  if (!transactionReceipt || !transactionHash) {
+    throw new Error(
+      `Could not find transaction receipt for transaction hash: ${transactionHash}`
+    )
+  }
 
   const deployment: Deployment = {
-    address: contractInstance.address,
+    address: contractAddress,
     abi: artifact.abi,
-    transactionHash: contractInstance.deployTransaction.hash,
+    transactionHash,
     implementation: implementation,
-    receipt: transactionReceipt,
+    // TODO: transactionReceipt type is not compatible with Deployment types
+    // receipt: transactionReceipt,
```

Comment:
Did you check if the receipt is used anywhere in the projects importing this library? I think we could get rid of it, if not. We are introducing breaking changes anyway so this is a good moment.

Reply:
Yup, I agree it's a good moment, other our projects are not using it and no other project will be forced to update that package (and tbh shouldn't unless it would upgrade ethers package as well) 204e0f4